app/services/service/base.rb

```diff
@@ -139,13 +139,15 @@ def initialize(name, method_name = name, class_name: nil, optional: nil)
 
       def call(instance, context)
         context[name] = super
-        raise ArgumentError, "Model not found" if !optional && context[name].blank?
+        if !optional && (!context[name] || context[name].try(:empty?))
```

Comment:
I _think_ we prefer the `&.` over `.try` 🤔 

```suggestion
        if !optional && (!context[name] || context[name]&.empty?)
```

Reply:
Not in that case because the safe navigation operator and `#try` have a subtle difference in their implementation: if the receiver is not `nil`, the safe navigation operator will try to call the provided method on it. On the other hand, `#try` will not call a non-existent method and will return `nil` :slightly_smiling_face: 

For example:
```ruby
model = User.first
!model || model.try(:empty?) # => false
!model || model&.empty? # => NoMethodError: undefined method `empty?' for an instance of User
```